### ISSUE TITLE
Auto fold when losing window focus

### ISSUE THREAD

**Issue by yzlnew** [equalsraf/neovim-qt#591]
Platform: Windows 1905

Steps to reproduce:

1. Open a file with folds
2. Open a fold
3. Switch to other app's window like Chrome, Neovim-qt loses focus
4. Folds close themselves. Have to open again.

Not happening on gVim or nvim in Terminal.

Here's my [vimrc](https://github.com/yzlnew/dotfiles/blob/master/Vim/vimrc)。

**Comment by jgehrig** [comment]
Hi @yzlnew

I cannot reproduce this issue. Please provide versions for `nvim` and `nvim-qt` along with a minimal `vimrc` file and explicit steps that reproduce the issue.

I suspect this is a problem with your vimrc configuration, and how it interacts with neovim.

**minimal.vim**
```
set foldmethod=indent
set foldlevel=0
```
**test_file**
```
asdf
    asdf
    asdf
    asdf
        asdfasdf
        asdfasdf
    asdf
asdf
```
**Test Scenario:**
1. Run command `nvim-qt -- -u minimal.vim test_file`.
2. Notice file is folded.
3. Open a fold with `zo`
4. Change focus to another application.
5. Return focus to neovim-qt.
6. All fold modifications remain.

**Comment by yzlnew** [comment]
Thanks. @jgehrig 
It seems that it's my vimrc causing the issue specificly in Neovim-qt. Is there any way of locating which plugin is closing folds when losing focus?

**Comment by equalsraf** [comment]
Ok, thanks @yzlnew. Resolving then.

**Comment by yzlnew** [comment]
@equalsraf @jgehrig Maybe it's still Neovim-qt related issue? Cause I don't see this behavior on `gVim` or terminal.

Here's a `minimal.vim` to test:

```viml
set nocompatible
set runtimepath^=~/.vim/plugged/vim-airline " point to your vim-airline
filetype plugin indent on
syntax on
set hidden
```

`test.vim`:

```viml
" {
" vim: foldmethod=marker foldlevel=0 foldmarker={,}
" It's a test
" }
```

Run

```
nvim-qt.exe -- -u .\minimal.vim .\test.vim
```

Use `zo` to open the folder. Click on other window.

**Comment by jgehrig** [comment]
@yzlnew I still cannot reproduce. It could be Windows specific? I will try that next time a have a Win10 machine available...

**Could you provide the output of `nvim --version`?**

**Do you know what version of nvim-qt you are using?**

Unfortunately, there is no --version option that I am aware of. We can track this down from where/how you installed.

I suspect this is an issue between your vimrc and how neovim interacts with a client using neovim-qt's parameters/configurations. Complete guess...

**Comment by yzlnew** [comment]
@jgehrig Yeah I'm on a Windows 10. I'll test on Manjaro later. Here's a minimal.vim without any plugins, the issue still persists.

By the way, I'm testing on nvim 0.4.0 and 0.5.0 nightly on my two machines. Both installed from `Scoop`, so Neovim-qt comes with https://github.com/neovim/neovim/releases

```
set nocompatible
au FocusLost * echo('Lost')
filetype plugin indent on
syntax on
set hidden
```

**Comment by jgehrig** [comment]
Thanks, I can now reproduce. I will investigate further...

**Comment by jgehrig** [comment]
Feel free to re-open, there is an issue here.

After investigating this further, the behavior is caused by the following:
https://github.com/equalsraf/neovim-qt/blob/be7051b1cc5b1dbc781c3b07d86d9293ca297d67/src/gui/shell.cpp#L1360-L1366

Triggering the `autocmd FocusLost` from nvim directly reproduces this behavior. It is not clear to me where to bug is, or why nvim doesn't have the same behavior when it triggers the autocmd FocusLost. How does the TUI trigger FocusLost?

**Example:**
1. Run command `nvim -u minimal.vim test.txt`
2. Open the fold with `zo`
3. Run nvim command `if exists('#FocusLost') | doautocmd FocusLost | endif`
4. Notice the fold closes, even though we are in nvim TUI.

**Related Issues:**
#329

**Related changes:**
957a5a503723a4f0bec503fabc1411921daff202
c1067b40e203b426e1f91d8355a38cb1f3c30eec

**Comment by jgehrig** [comment]
@yzlnew Please re-open, and let me know if this Pull Request resolves your issue.

Oni does not reproduce because it triggers with `<nomodeline>` I have added the same pattern here.

Do you know if this behavior is modeline specific? Is there a case that reproduces, and does not use modelines?

I'm not sure I fully understand what is going on here. Maybe someone with better knowledge of noevim can comment?

**Comment by yzlnew** [comment]
@jgehrig I can't re-open the issue, maybe because it's closed bt @equalsraf ?
I'm on vacation now and have no access to a Winodows device. I'll let you know asap if `<nomodeline>` works.

**Comment by jgehrig** [comment]
Thanks for the re-open @equalsraf

This isn't a Windows specific issue, I can reproduce on Linux too. It should fix your test case, but I'm not sure about the original case. The test case has a modeline, and I'm not sure if that is the cause in your original case...

@yzlnew No rush on this, enjoy your vacation! :smile:

**Comment by yzlnew** [comment]
It's been fixed. 🍺
Closing now.
